hive/src/main/java/org/apache/iceberg/hive/legacy/HiveCatalogWithLegacyReadFallback.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hive.legacy;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A {@link HiveCatalog} which uses falls back to using Hive metadata to read tables when Iceberg metadata is not
+ * available. If the table is read through Hive metadata, features like time travel, snapshot isolation and incremental
+ * computation are not supported along with any WRITE operations to either the data or metadata.
+ */
+public class HiveCatalogWithLegacyReadFallback extends HiveCatalog {
```

Comment:
Legacy is a loaded word. HiveCatalogWithReadFallback is just as descriptive and takes up less screen real estate.

Reply:
As discussed today over call we will simply be renaming it to LegacyHiveCatalog and let caller handle fallbacks.

Reply:
I think the point was about the word "Legacy", and the fact that LegacyHiveCatalog does not tell whether Legacy refers to Hive or the Catalog. One who is not familiar with the intention will interpret the Catalog is Legacy.

Reply:
I will keep it as LegacyHiveCatalog till we come up with a term that everyone is satisfied with. The usage of this catalog is internal and we have enough Javadoc to clarify its intention. It is also really Legacy as once Iceberg metadata is rolled out, this will no longer be needed.